### DISCUSSION TITLE
Do we really need to type `cpp2::unsafe_narrow` instead of just `unsafe_narrow`?

### DISCUSSION THREAD

**Discussion by DyXel** [hsutter/cppfront#1122] [Q&A] [answered]
Title. I know we want to discourage narrowing casts where possible, but usually these utility functions are lowered with the `cpp2::` prepended (like `new`). Unfortunately its pretty common for C APIs to use `int` where `size_t` or similar would have made more sense 😦 

Another reason I ask is because in all languages I have used, cast utilities are "built-ins" they are not namespaced.

**Comment by gregmarr** [comment]
> Do we really need to type cpp2::unsafe_narrow instead of just unsafe_narrow? #1122

No, you don't have to if you don't want to.

```
using cpp2::unsafe_narrow;

main: () -> int = { return unsafe_narrow<int>(0); }
```

**Reply by DyXel** [reply]
Alright, I guess I mean _by default_.

**Comment by hsutter** [comment, marked as answer]
In the case of things like `new` which are safe and recommended, I go out of my way to make people not have to write `cpp2::`.

For `unsafe_narrow`, noisiness is a feature I think, even if it's needed often for C APIs.
